### PR TITLE
feat: separate groups into major/minor/patch

### LINKED ISSUE #2425: Grouping + separateMajorMinor
When dependencies are grouped, this effectively sets `separateMajorMinor` functionality to `false` implicitly. e.g. if you have a group `React packages` then the `branchName` becomes `renovate/react-packages` and therefore whatever is the "latest" package update is always used, whether major or minor. This can result in major and minor packages being mixed together.

I think that we already have `separateMajorMinor` and it's not clear why we *have* to combine together major and minor without any option to control that.

There are two parts to this solution that I can think of:
1. Separating groups into major and minor
2. Distinguishing them via PR title (e.g. add a `(major)` suffix

### PR DESCRIPTION
**Pull request by rarkins**
Previously if grouping, all major/minor updates within that group were combined into one. Instead, we now honour the “separateMajorMinor”, "separateMinorPatch", and "separateMultipleMajor" settings and keep the groups separate if necessary.

For maximum compatibility with existing PRs, we name branches like `renovate/group-name` whenever possible and only name them like `renovate/major-group-name` or `renovate/patch-group-name` if major or patch are found.

Closes #2425

### CODE REVIEW

**Comment by hbetts** [comment]
Does a separate issue need to be filed for `separateMinorPatch`?

File: test/workers/repository/process/lookup/index.spec.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ describe('manager/npm/lookup', () => {
         .reply(200, qJson);
       expect((await lookup.lookupUpdates(config)).updates).toMatchSnapshot();
     });
-    it('returns only one update if grouping', async () => {
+    it('returns two updates if grouping but separateMajorMinor=true', async () => {
```
Comment (hbetts) [inline test/workers/repository/process/lookup/index.spec.js]: Is there a test for when there are two updates and `separateMajorMinor=false`?

Comment (rarkins) [reply test/workers/repository/process/lookup/index.spec.js]: @hbetts added now

**Comment by rarkins** [comment]
Now also supporting `separateMinorPatch=true`.

**Comment by rarkins** [comment]
This PR now supports `separateMajorMinor`, `separateMinorPatch` and `separateMultipleMajor`.

It's optimised for the default case of `separateMajorMinor=true` and the others are false. In that case, branches are named like `renovate/major-foo` and `renovate/foo` instead of `renovate/minor-foo`. This will cause the least disturbance to existing PRs once this is rolled out. If `separateMinorPatch=true` then you'll see `renovate/minor-foo` and `renovate/patch-foo`.

**Pull request merged**